crowbar_framework/app/models/api/node.rb

```diff
@@ -15,6 +15,56 @@
 #
 
 module Api
-  class Node
+  class Node < Tableless
+    def repocheck(options = {})
+      addon = options.fetch(:addon, "os")
+      features = ["os"]
+      features.push(addon) if addon
+
+      {}.tap do |ret|
+        ret[addon] = {
+          "available" => true
+        }
+        platform = Api::Upgrade.new.target_platform(platform_exception: addon)
+
+        features.each do |feature|
+          node_architectures(addon: addon).each do |architecture|
+            unless ::Crowbar::Repository.provided_and_enabled?(feature,
```

Comment:
Does the `provided_and_enabled` take "os" value for a feature? Was that added in some PR I do not see?
And when do we check for presence of Cloud repository?